File: shortest_word_distance_1.py
#Time Complexity : O(N)  #N is words in dictionary
#Space Complexity : O(1) 
class Solution:
    def shortestDistance(self, wordsDict: List[str], word1: str, word2: str) -> int:
        p1 = -1
        p2 = -1
        mind = float('inf')
        for index,s in enumerate(wordsDict):
            if s == word1:
                p1 = index
            elif s == word2:
                p2 = index
            
            if p1 !=-1 and p2!=-1:
                mind = min(mind,abs(p1-p2))
        return mind

File: assignments/src/ShortestWordDistance_1.java
// Approach: Two pointers and Min for each pair. Return the minimum at end.
// Time: O(n)
// Space: O(1)

class ShortestWordDistance_1 {
    public int shortestDistance(String[] wordsDict, String word1, String word2) {
        int p1 = -1, p2 = -1, min = Integer.MAX_VALUE;

        for (int i = 0; i<wordsDict.length; i++) {
            if (wordsDict[i].equals(word1)) {
                p1 = i;
            }
            if (wordsDict[i].equals(word2)) {
                p2 = i;
            }
            if (p1 != -1 && p2 != -1) {
                min = Math.min(min, Math.abs(p1-p2));
            }
        }
        return min;
    }
}